### PR TITLE
Upgrade to use OpenTelemetry version 0.13.0

### PR DESCRIPTION
**Pull request by seh**
Adapt the Honeycomb OpenTelemetry exporter to use the OpenTelemetry [_opentelemetry-go_](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v0.13.0) and [_opentelemetry-go-contrib_](https://github.com/open-telemetry/opentelemetry-go-contrib/releases/tag/v0.13.0) libraries as of their coordinated releases at version 0.13.0.

Update the example client and server programs to demonstrate use of trace propagation, including baggage. For the server program, augment it to accept its listening address and port as command-line flags, use an HTTP handler multiplexer to accommodate additional URL paths, and to shut down cleanly upon receiving INT and TERM signals.

While we're here, address all complaints from _golint_, _go vet_, and _staticcheck_.

### CODE REVIEW

**Comment by seh** [comment]
Note that open-telemetry/opentelemetry-go-contrib#399 tracks the less-than-ideal span hierarchy created by the example client's HTTP requests.

**Review by lizthegrey — APPROVED** [review]

File: example/client/client.go [hunk 1]
```diff
@@ -9,29 +9,37 @@ import (
...
+			// For the demonstration, use sdktrace.AlwaysSample sampler to sample all traces.
+			// In a production application, use sdktrace.ProbabilitySampler with a desired
+			// probability.
+			DefaultSampler: sdktrace.AlwaysSample(),
+		})
 	global.SetTracerProvider(tp)
+	global.SetTextMapPropagator(otel.NewCompositeTextMapPropagator(
+		propagators.TraceContext{},
+		propagators.Baggage{}))
+	return bsp.Shutdown
```
Comment (seh) [inline example/client/client.go:42]: See open-telemetry/opentelemetry-specification#1073 for how this situation might change soon. My comment https://github.com/open-telemetry/opentelemetry-specification/issues/1073#issuecomment-706458748 addresses my use of `NewBatchSpanProcessor` above, as opposed to using `WithBatcher`.

**Review by MikeGoldsmith — APPROVED** [review]
This is fantastic, thanks @seh!

**Comment by MikeGoldsmith** [comment]
Just had to close/re-open to trigger CI 👍

File: example/client/client.go (hunk 1 above)
Comment (seh) [reply example/client/client.go:42]: open-telemetry/opentelemetry-go#1227 just merged, allowing shutting down a `TracerProvider`, in trun shutting down its registered span processors. Once we have release available that includes that new method, I'll propose an update to simplify this example.

**Pull request merged**